File: aoj/volume100/ocaml/A10010.ml
let rec loop () =
    let (a, op, b) = Scanf.scanf "%d %s %d\n" (fun a op b -> (a, op, b)) in
    match op with
    | "?" -> ()
    | _   -> (
        let result = match op with
        | "+" -> a + b
        | "-" -> a - b
        | "*" -> a * b
        | "/" -> a / b
        | _   -> 0
        in
        Printf.printf "%d\n" result ;
        loop () ;
    )
in
loop ()

File: aoj/volume100/scala/A10010.scala
object Main {
  def main(args: Array[String]) {
    loop
  }

  def loop {
    val input = scala.io.StdIn.readLine split " "
    val a  = input(0).toInt
    val op = input(1)
    val b  = input(2).toInt
    if (op == "?") {
      return
    }

    val result = op match {
      case "+" => a + b
      case "-" => a - b
      case "*" => a * b
      case "/" => a / b
      case _   => 0
    }
    println(result)
    loop
  }
}
